### DISCUSSION TITLE
container with two networks experiences intermittent packet loss

### DISCUSSION THREAD

**Discussion by 1player** [containers/netavark#828] [Q&A] [answered]
I'd like some advice on what I am doing wrong. I have a container attached to two podman networks (simply created with `podman network create`) that is losing every other packet. This is when pinging a known IP from inside the network namespace

```
[root@bernard ~]# ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1) 56(84) bytes of data.
64 bytes from 1.1.1.1: icmp_seq=1 ttl=57 time=5.47 ms
64 bytes from 1.1.1.1: icmp_seq=3 ttl=57 time=5.47 ms
64 bytes from 1.1.1.1: icmp_seq=5 ttl=57 time=5.44 ms
64 bytes from 1.1.1.1: icmp_seq=7 ttl=57 time=5.45 ms
```

Notice the icmp_seq numbers. Of course this happens only in the container. I notice that each network I have attached to it has its own default gateway:

```
[root@bernard ~]# ip route list
default via 10.89.2.1 dev eth1 proto static metric 100
default via 10.89.3.1 dev eth0 proto static metric 100
10.89.2.0/24 dev eth1 proto kernel scope link src 10.89.2.12
10.89.3.0/24 dev eth0 proto kernel scope link src 10.89.3.12
```

Which might indicate that one packet is sent round-robin on each IP? tcpdump isn't really helping here, as I am seemingly unable to intercept all traffic to figure out what's going on.

The question is: what am I doing wrong? Why am I losing every other packet? How am I supposed to attach two different networks to a container (I don't even need them to be default routes, they are used for intra-container communication)

**Comment by Luap99** [comment, marked as answer]
Do you have `sysctl net.ipv4.conf.default.rp_filter` set to 1? You have to set it to 2 then restart the container.

**Reply by 1player** [reply]
Shouldn't podman/netavark do that automatically for me when I attach two networks? Or do you mean I should set that on the host?

**Reply by cornmail** [reply]
Thank you for the suggestions regarding the packet loss issue. I'm curious, though—what alternatives do we have in a rootless environment for managing network configurations or troubleshooting similar issues? Any insights would be greatly appreciated!
